### PR TITLE
222837 Add account management commands to the CLI tool: add, list, remove, activate

### PR DESCRIPTION
**Pull request by d3rky**

### CODE REVIEW

File: swo/mpt/cli/core/console.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        TaskProgressColumn(),
+        TimeElapsedColumn(),
+        console=console,
+        expand=True,
+    )
+
+
+@contextlib.contextmanager
+def status_progress() -> Generator[tuple[Status, Progress], None, None]:
+    status = Status("")
```
Comment (ffaraone) [inline swo/mpt/cli/core/console.py:85]: Just for visual consistency I would suggest `_status`.

File: swo/mpt/cli/core/alias_group.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    are same commands
+    """
+
+    def get_command(self, ctx, cmd_name):
+        rv = click.Group.get_command(self, ctx, cmd_name)
+        if rv is not None:
+            return rv
+
+        matches = [name for name in self.list_commands(ctx) if name == f"{cmd_name}s"]
```
Comment (ffaraone) [inline swo/mpt/cli/core/alias_group.py:19]: I would suggest to use https://pypi.org/project/inflect/ here.

File: swo/mpt/cli/core/mpt/flows.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+@wrap_http_error
+def get_token(mpt_client: MPTClient, token_id: str) -> Token:
+    """
+    Retrieves API Token from the MPT Platform
+    """
+    response = mpt_client.get(f"/accounts/api-tokens/{token_id}")
+    response.raise_for_status()
+
+    return Token.model_validate(response.json())
```
Comment (ffaraone) [inline swo/mpt/cli/core/mpt/flows.py:15]: Why not `return Token(**response.json())` ?

File: swo/mpt/cli/core/accounts/flows.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    """
+    Extracts Account from the MPT Token
+    """
+    return Account(
+        id=token.account.id,
+        name=token.account.name,
+        type=token.account.type,
+        secret=secret,
+        environment=environment,
+        is_active=True,
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/flows.py:21]: I would suggest to just call it `active`. The `is_` prefix makes me think on a function.

File: swo/mpt/cli/core/accounts/models.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from pydantic import BaseModel
+
+
+class Account(BaseModel):
+    id: str
+    name: str
+    type: str
+    secret: str
+    environment: str
+    is_active: bool = False
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/models.py:10]: I would suggest to just call it `active`. The `is_` prefix makes me think on a function.

File: swo/mpt/cli/core/accounts/flows.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+
+def get_accounts_file_path() -> Path:
+    """
+    Returns accounts file path
+    """
+    return Path.home() / ".swocli" / "accounts.json"
+
+
+def is_account_exists(accounts: list[Account], account: Account) -> bool:
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/flows.py:54]: I would suggest `does_account_exist`.

File: swo/mpt/cli/core/accounts/app.py [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+        str, typer.Argument(help="SoftwareOne Marketplace API Token secret")
+    ],
+    environment: Annotated[
+        str, typer.Option("--environment", "-e", help="URL to the API for environment")
+    ] = "https://api.platform.softwareone.com/v1",
+):
+    """
+    Add account to work with the SoftwareOne Marketplace
+    """
+    with status_progress() as (status, _):
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/app.py:40]: Please take a look also to https://rich.readthedocs.io/en/latest/_modules/rich/status.html#Status and to https://github.com/softwareone-platform/swo-adobe-vipm-extension/blob/main/devmock/mockgendata.py for an alternative way of rendering an undetermined progress.

File: swo/mpt/cli/core/accounts/app.py [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        elif account_type == "Operations":
+            return f"[white]{account_type}"
+        elif account_type == "Client":
+            return f"[magenta]{account_type}"
+        else:
+            return account_type
+
+    def _wrap_active(is_active: bool) -> str:  # pragma: no cover
+        if is_active:
+            return "[red bold]\u2714"
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/app.py:201]: I would prefer a green for a tick symbol that means `active`l.

File: swo/mpt/cli/core/accounts/app.py [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+        if is_active:
+            accounts = list(filter(lambda a: a.is_active, accounts))
+
+    if not accounts:
+        console.print("No account found")
+        raise typer.Exit(code=0)
+
+    table = _account_table("Available accounts")
+    table = _list_accounts(table, accounts, is_secret=False)
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/app.py:169]: Why you are calling it with `is_secret=False` ?

File: swo/mpt/cli/core/accounts/app.py [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+        write_accounts(accounts_file, accounts)
+
+    table = _account_table("Account has been deleted")
+    table = _list_accounts(table, [account])
+    console.print(table)
+
+
+@app.command(name="list")
+def list_accounts(
+    is_active: Annotated[
```
Comment (ffaraone) [inline swo/mpt/cli/core/accounts/app.py:149]: `active_only` ?

File: swo/mpt/cli/core/accounts/app.py [hunk 11]
```diff
@@ -0,0 +1,225 @@
+import typer
+from rich.table import Table
+from swo.mpt.cli.core.console import console, status_progress
+from swo.mpt.cli.core.errors import AccountNotFoundError, MPTAPIError
+from swo.mpt.cli.core.mpt.client import MPTClient
+from swo.mpt.cli.core.mpt.flows import get_token
+from typing_extensions import Annotated
```
[the remark this answers is not in the document]
Comment (d3rky) [reply swo/mpt/cli/core/accounts/app.py:7]: 👍

File: swo/mpt/cli/swocli.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import typer
+from swo.mpt.cli.core.accounts import app as accounts_app
+from swo.mpt.cli.core.alias_group import AliasTyperGroup
+from swo.mpt.cli.core.console import show_banner
+
+app = typer.Typer(cls=AliasTyperGroup, callback=show_banner, help="Test sdfadsfds")
```
[the remark this answers is not in the document]
Comment (d3rky) [reply swo/mpt/cli/swocli.py:6]: ahahaha, exactly!

**Review by ffaraone — APPROVED** [review]

**Review by ffaraone — APPROVED** [review]

**Pull request merged**